signer/src/storage/postgres.rs

```diff
@@ -1411,19 +1414,28 @@ impl super::DbRead for PgStore {
                 deposit_requests AS deposit_req 
                     ON deposit_req.txid = swept_deposit.deposit_request_txid
                     AND deposit_req.output_index = swept_deposit.deposit_request_output_index
-            -- TODO: The following left join is incorrect, we need to check that 
-            -- the completed deposit event is in a Stacks block that is linked 
-            -- to the canonical Bitcoin chain (#559).
             LEFT JOIN 
                 completed_deposit_events AS cde
                     ON cde.bitcoin_txid = deposit_req.txid
                     AND cde.output_index = deposit_req.output_index
-            WHERE
-                cde.bitcoin_txid IS NULL
+            LEFT JOIN
+                s_blocks
+                    ON s_blocks.block_hash = cde.block_hash
+            GROUP BY
+                bc_trx.txid
+              , bc_trx.block_hash
+              , bc_blocks.block_height
+              , deposit_req.txid
+              , deposit_req.output_index
+              , deposit_req.recipient
+              , deposit_req.amount
+            HAVING
+                COUNT(s_blocks.block_hash) = 0
```

Comment:
I think we can simplify this to
```sql
WHERE s_blocks.block_hash IS NULL
```

Reply:
This would actually be wrong, and an explanation why can be found in https://github.com/stacks-network/sbtc/pull/747#discussion_r1826660102.